### ISSUE TITLE
Experimental support for materials from other mods.

### ISSUE THREAD

**Issue by vlad0-0** [SilentChaos512/Silent-Gear#472]
## What Should Be Added or Changed?
If a gear is found in another mod that is created from the same type of material and has a recipe similar to vanilla, then it will be added to the silent gear materials table. The material characteristics will be based on the properties and characteristics of the equipment from this mod.
(Or at least add the ability to manually select the material to be included in the table using a convenient in-game interface, without having to search for the necessary configuration files.)
## Why Should This Be Done?
If this task is not too difficult, it will allow to diversify the experience even more.

**Comment by vlad0-0** [comment]
If there is a bar/ingot in the name of the material, it is most likely metal. If there are planks in the name, it is a wooden material. If it is a block, it is most likely a stone. If it's an item, it's most likely a gem.

**Comment by vlad0-0** [comment]
If there is a bar/ingot in the name of the material, it is most likely metal. If there are planks in the name, it is a wooden material. If it is a block, it is most likely a stone. If it's an item, it's most likely a gem.

**Comment by SilentChaos512** [comment]
Why was this issue closed? It can be reopened if you closed it accidentally.

**Comment by vlad0-0** [comment]
Yes, I accidentally closed it

**Comment by SilentChaos512** [comment]
Okay. This is definitely an interesting idea. Getting the stats from the tools would be simple enough. Scanning the recipes is doable, but might be too expensive to do every time materials are loaded. So perhaps I could add a command that generates a data pack instead?

**Comment by vlad0-0** [comment]
Yes, or you can scan one time and save the results in an additional expandable table for the next loads if possible. If the list of  mods does not change, then the cache of materials will remain the same. The remaining mods can be marked as scanned so as not to repeat this procedure.
Perhaps it will be a little better than creating a datapack for each generated world.

**Comment by vlad0-0** [comment]
You can also make a check for the drop of ores. If the ore drops itself, then it is most likely metal. If the ore drops an item and it is part of the equipment, then it is most likely a gem. It seems more complicated in my opinion. But this option can complement the previous one.

**Comment by Tenchu98** [comment]
I know this is nearly a year old, but I love the idea of a command to generate a datapack for potential materials in a pack.

**Comment by vlad0-0** [comment]
I am not well versed in how this could be optimized in the future, but I will be glad if it is added in any form in the early stages. It confuses me that perhaps because of my comments, Silent has lost the desire to do this at that time

**Comment by SilentChaos512** [comment]
I have very little time for coding anymore. It's not that I don't want to do this, I just haven't had the chance to. I will try to add this eventually.
